source/constructing-lattices.md

```diff
@@ -0,0 +1,306 @@
+
+%---------------------------------------------------------------------------------------------------
+(s:direction)=
+### Direction reversal
+
+The optional `direction` component of `item` can be used for true [direction reversal](#s:ref.construct).
+Possible values are `+1` and `-1`. The Default is `+1` which represents an unreversed element
+or BeamLine. BeamLine reversal involves both reversed order of the line and direction reversal of
+the individual line items. Example:
+```{code} yaml
+BeamLine:
+  name: lineA
+  line:
+    - item: 
+        direction: -1
+        name: lineB
+
+BeamLine:
+  name: lineB
+  line:
+    - item: ele1
+    - item:
+        direction: -1
+        name: ele2
+```
+The expanded `lineA` has elements:
+```{code} yaml
+ele2, -ele1
+```
+where the negative sign here indicates that `ele1` is reversed in direction. 
+Notice that `ele2` is not reversed since the reversal of a reversed element results
+in an element that is unreversed.
+
+%---------------------------------------------------------------------------------------------------
+(s:placement)=
+### Line item placement
+
+```{figure} figures/superimpose.svg
+:width: 80%
+:name: f:superposition
+
+Positioning of a line item (which may be a lattice element or BeamLine) with respect to 
+a reference line item when there is an explicit placement.
+```
+
+If the longitudinal placement of a line `item` is not specified, as is the case with the above
+examples, a line `item` is placed so the upstream end of the item is flush with the downstream end
+of the preceding item as explained in the [Branch Coordinates Construction](#s:ref.construct) section.
+
+Explicit longitudinal placement of a line `item` uses a `placement` component. 
+The components of `placement` are:
+```{code} yaml
+offset            # Longitudinal offset of the line item.
+origin            # Line item origin point.
+reference         # Reference line item.
+reference_origin  # Reference line item origin point.
+```
+Figure {numref}`f:superposition` shows how a line item is positioned with respect to a reference line item. 
+If the reference element is not specified, the beginning of the line is used. In this case, 
+a `referece_origin` may not be specified.
+
+Example:
+```{code} yaml
+BeamLine:
+  line:
+    - item: thingA
+    - item
+        name: this_line
+        placement:
+          offset = 37.5
+          reference: thingA
+          reference_origin: DOWNSTREAM_END
+          origin: REFERENCE_ELEMENT
+        ...
+    ...
+```
+
+The `reference_origin` is the reference point on the reference line element and `origin` is the
+reference point on the element being positioned. These switches may take the values:
+```{code} yaml
+UPSTREAM_END
+CENTER
+DOWNSTREAM_END
+REFERENCE_ELEMENT    # Used with sublines that define a reference element.         
+```
+Note: Lattice elements are allowed to overlap but it should be kept in mind that this 
+may hinder portability with other programs.
```

Comment:
Replace "may hinder portability with other programs" with, e.g., "may not be supported by some beam tracking software"?

Reply:
Changed to
```
Note: Lattice elements are allowed to overlap but it should be kept in mind that 
some programs will not be able to handle overlapping fields.
```